build is failing
```
======================================================================
ERROR: test_ledger_balances_indicator (corehq.apps.userreports.tests.test_indicators.LedgerBalancesIndicatorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/tests/test_indicators.py", line 489, in test_ledger_balances_indicator
    values = indicator.get_values(doc, EvaluationContext(doc, 0))
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/indicators/__init__.py", line 97, in get_values
    return [val for ind in self.indicators for val in ind.get_values(item, context)]
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/indicators/__init__.py", line 82, in get_values
    return [ColumnValue(self.column, self.getter(item, context))]
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/indicators/factory.py", line 107, in get_stock_state
    sql_product__code=product_code,
TypeError: mock_getter() got an unexpected keyword argument 'sql_product__code'
----------------------------------------------------------------------
```

--- a/corehq/apps/userreports/indicators/factory.py
+++ b/corehq/apps/userreports/indicators/factory.py
@@ -5,11 +5,12 @@ from corehq.apps.userreports.filters import SinglePropertyValueFilter, CustomFil
 from corehq.apps.userreports.filters.factory import FilterFactory
 from corehq.apps.userreports.indicators import BooleanIndicator, CompoundIndicator, RawIndicator, Column
 from corehq.apps.userreports.indicators.specs import (
-    RawIndicatorSpec,
-    ChoiceListIndicatorSpec,
     BooleanIndicatorSpec,
-    IndicatorSpecBase,
+    ChoiceListIndicatorSpec,
     ExpressionIndicatorSpec,
+    IndicatorSpecBase,
+    LedgerBalancesIndicatorSpec,
+    RawIndicatorSpec,
 )
 
 
@@ -85,6 +86,30 @@ def _build_choice_list_indicator(spec, context):
     return CompoundIndicator(base_display_name, choice_indicators)
 
 
+def _build_ledger_balances_indicator(spec, context):
+    wrapped_spec = LedgerBalancesIndicatorSpec.wrap(spec)
+    base_display_name = wrapped_spec.display_name
+
+    def _construct_display(product_code):
+        return '{base} ({product_code})'.format(base=base_display_name,
+                                                product_code=product_code)
+
+    def _construct_column(product):
+        return '{col}_{product}'.format(col=spec['column_id'], product=product)
+
+    product_indicators = [
+        RawIndicator(
+            display_name=_construct_display(product_code),
+            column=Column(
+                id=_construct_column(product_code),
+                datatype="decimal",
+            ),
+            getter=wrapped_spec.getter,
+        ) for product_code in wrapped_spec.product_codes
+    ]
+    return CompoundIndicator(base_display_name, product_indicators)
+
+
 def _build_repeat_iteration_indicator(spec, context):
     return RawIndicator(
         "base document iteration",
@@ -117,9 +142,10 @@ class IndicatorFactory(object):
         'choice_list': _build_choice_list_indicator,
         'count': _build_count_indicator,
         'expression': _build_expression_indicator,
+        'inserted_at': _build_inserted_at,
+        'ledger_balances': _build_ledger_balances_indicator,
         'raw': _build_raw_indicator,
         'repeat_iteration': _build_repeat_iteration_indicator,
-        'inserted_at': _build_inserted_at,
     }
 
     @classmethod

--- a/corehq/apps/userreports/indicators/factory.py
+++ b/corehq/apps/userreports/indicators/factory.py
@@ -104,7 +104,7 @@ def _build_ledger_balances_indicator(spec, context):
                 return StockState.objects.get(
                     section_id=wrapped_spec.ledger_section,
                     case_id=doc['_id'],
-                    product_id=product_code,
+                    sql_product__code=product_code,
                 ).stock_on_hand
             except StockState.DoesNotExist:
                 return 0

--- a/corehq/apps/userreports/indicators/specs.py
+++ b/corehq/apps/userreports/indicators/specs.py
@@ -97,3 +97,12 @@ class ChoiceListIndicatorSpec(PropertyReferenceIndicatorSpecBase):
 
     def get_operator(self):
         return IN_MULTISELECT if self.select_style == 'multiple' else EQUAL
+
+
+class LedgerBalancesIndicatorSpec(IndicatorSpecBase):
+    type = TypeProperty('ledger_balances')
+    product_codes = ListProperty(required=True)
+    ledger_section = StringProperty()
+
+    def getter(self, item, context):
+        return 32

--- a/corehq/apps/userreports/tests/test_indicators.py
+++ b/corehq/apps/userreports/tests/test_indicators.py
@@ -1,9 +1,11 @@
 from copy import copy
 from decimal import Decimal
+from mock import patch
 from django.test import SimpleTestCase
 from corehq.apps.commtrack.models import StockState
 from corehq.apps.userreports.exceptions import BadSpecError
 from corehq.apps.userreports.indicators.factory import IndicatorFactory
+from corehq.apps.userreports.specs import EvaluationContext
 
 
 class SingleIndicatorTestBase(SimpleTestCase):
@@ -454,6 +456,14 @@ class LedgerBalancesIndicatorTest(SimpleTestCase):
                 "ghi",
             ]
         }
+        self.stock_states = {}
+        for val, case_id, product_id in [
+            (32, 'case1', 'abc'),
+            (85, 'case1', 'def'),
+            (11, 'case1', 'ghi'),
+        ]:
+            stock_state = self._make_stock_state(val, case_id, product_id)
+            self.stock_states[(case_id, product_id)] = stock_state
 
     @staticmethod
     def _make_stock_state(value, case_id, product_id, section_id='soh'):
@@ -464,12 +474,19 @@ class LedgerBalancesIndicatorTest(SimpleTestCase):
             section_id=section_id,
         )
 
-    def test_ledger_balances_indicator(self):
+    def mock_getter(self, section_id, case_id, product_id):
+        if (case_id, product_id) in self.stock_states:
+            return self.stock_states[(case_id, product_id)]
+        else:
+            raise StockState.DoesNotExist()
+
+    @patch('corehq.apps.userreports.indicators.factory.StockState.objects')
+    def test_ledger_balances_indicator(self, stock_states):
         indicator = IndicatorFactory.from_spec(self.spec)
-        self._make_stock_state(32, 'case1', 'abc')
-        self._make_stock_state(85, 'case1', 'def')
-        self._make_stock_state(11, 'case1', 'ghi')
-        values = indicator.get_values({'_id': 'case1'})
+        stock_states.get = self.mock_getter
+
+        doc = {'_id': 'case1'}
+        values = indicator.get_values(doc, EvaluationContext(doc, 0))
 
         self.assertEqual(
             [(val.column.id, val.value) for val in values],

